The second dataset that we will look at was posted by silvia and details the passengers aboard the *Titanic*.

In [29]:
import pandas as pd
df = pd.read_csv("https://raw.githubusercontent.com/amandamichelle27/IS362-Project2/master/Titanic.csv")
df.head()

Unnamed: 0                                           Name PClass    Age  \
0           1                   Allen, Miss Elisabeth Walton    1st  29.00   
1           2                    Allison, Miss Helen Loraine    1st   2.00   
2           3            Allison, Mr Hudson Joshua Creighton    1st  30.00   
3           4  Allison, Mrs Hudson JC (Bessie Waldo Daniels)    1st  25.00   
4           5                  Allison, Master Hudson Trevor    1st   0.92   

      Sex  Survived  SexCode  
0  female         1        1  
1  female         0        1  
2    male         0        0  
3  female         0        1  
4    male         1        0

By looking at the data here we can see that a couple of the colums are indeed unnecessary. However, since Pandas stores the data colum by colum, we don't waste any time computing with the colums that aren't being used. But for the sake of just making the data look nicer we can go ahead and remove the unnecessary colums. 

In [30]:
df.drop(["Unnamed: 0", "SexCode"], axis=1, inplace=True)
df.head()

Name PClass    Age     Sex  \
0                   Allen, Miss Elisabeth Walton    1st  29.00  female   
1                    Allison, Miss Helen Loraine    1st   2.00  female   
2            Allison, Mr Hudson Joshua Creighton    1st  30.00    male   
3  Allison, Mrs Hudson JC (Bessie Waldo Daniels)    1st  25.00  female   
4                  Allison, Master Hudson Trevor    1st   0.92    male   

   Survived  
0         1  
1         0  
2         0  
3         0  
4         1

The next step will be to search for any missing data.

In [31]:
df.isnull().sum()

Name          0
PClass        0
Age         557
Sex           0
Survived      0
dtype: int64

In [32]:
df.shape

(1313, 5)

Here we see that the only colum with missing data is "Age". Out of 1313 rows, 557 of them are missing an age entry. But since pandas will automatically ignore the missing entries when calculating, we don't need to do anything about this. We can go ahead and calculate the mean and median of the passenger ages.

In [33]:
print(df["Age"].median())
print(df["Age"].mean())

28.0
30.397989417989415


Next we can look at which sex had the highest amount of survivors. We group the data by "Survived" and "Sex", then only have it count the entries in the Name colum. This way it only returns the data that we need, however having the count done by any of the other colums other except for age (because there are missing values) would have worked.  

In [34]:
df.groupby(["Survived", "Sex"])["Name"].count()

Survived  Sex   
0         female    154
          male      709
1         female    308
          male      142
Name: Name, dtype: int64

This shows us that over twice as many women survived than men. We can use the same method to look out which passenger class had the most survivors. 

In [35]:
df.groupby(["Survived", "PClass"])["Name"].count()

Survived  PClass
0         *           1
          1st       129
          2nd       160
          3rd       573
1         1st       193
          2nd       119
          3rd       138
Name: Name, dtype: int64

Doing this we can see that the first class passengers are the only group that had more passengers survive than die, as well as had the highest number of survivors overall. 